pajbot/apiwrappers/twitch/helix.py

```diff
@@ -602,3 +602,17 @@ def get_channel_emotes(
             expiry=60 * 60,
             force_fetch=force_fetch,
         )
+
+    def send_chat_announcement(self, bot_id: str, body: Dict[str, str], authorization, channel_id: str = None):
+        if not body:
+            log.error("Invalid call to send_chat_announcement, missing query parameter. message must be specified")
```

Comment:
Don‘t do this. Either just don‘t do this check at all (I‘m sure Twitch will send an error response anyways), or at least raise an exception. Especially returning `False` is extremely bad practice here.

Reply:
This was a direct copy from the `modify_channel_information` function. Given we already check this in the `announce` function, this is likely not something that will be used. In saying that, I don't think we should just leave it to Twitch to respond to no message - it's a waste of an API call, especially when the check is already performed in the current version of the `announce` function.

Reply:
The `modify_channel_information` needs to be rewritten in that case. Very poor API design that's most certainly not present in all the other apiwrapper functions in here or any of the other apiwrappers.